pkg/reconciler/reconciler.go

```diff
@@ -436,6 +449,7 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (res ctrl.Result, err error) {
 		// The decision made for now is to leave the finalizer in place, so that the user can intervene and try to
 		// resolve the issue, instead of the operator silently leaving some dangling resources hanging around after the
 		// CR is deleted.
+		// todo: check : Is this the same behaviour in SDK or it was introduced with the ActionClient?
 		return ctrl.Result{}, err
```

Comment:
Question about finalizer note: Is this the same behaviour in SDK or it was introduced with the ActionClient?

Reply:
I _think_ this is pretty much the same behavior as in SDK. Think of the ActionClient here like the release manager in the SDK implementation. [We don't remove the finalizer if there's an error getting the release manager](https://github.com/operator-framework/operator-sdk/blob/ca0e78d90e69b7cefa0d7f315d3b027b82b8e858/pkg/helm/controller/reconcile.go#L90-L94).